### PR TITLE
Feature/rnp 126 ios

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

File: ios/Handlers/RegistrationHandler.swift [hunk 1]
```diff
@@ -76,14 +74,19 @@ extension RegistrationHandler : RegistrationConnectorToHandlerProtocol {
...
+            throw WrapperError.actionNotAllowed(description: "Canceling the custom registration right now is not allowed. Registration is not in progress or pin creation has already started.")
+        }
+        customRegistrationChallenge.sender.cancel(customRegistrationChallenge)
+        handleDidFailToRegister()
     }
 
     func cancelBrowserRegistration() throws {
         guard let browserRegistrationChallenge = self.browserRegistrationChallenge else {
-            throw WrapperError.registrationNotInProgress
+            throw WrapperError.actionNotAllowed(description: "Canceling the browser registration right now is not allowed. Registration is not in progress or pin creation has already started.")
```
Comment (kfa-onewelcome) [inline ios/Handlers/RegistrationHandler.swift:86]: that description is used twice, can you declare it static and used in both places?

File: ios/OneginiWrapperErrors.swift [hunk 2]
```diff
@@ -7,8 +7,10 @@ enum WrapperError: LocalizedError {
     case registrationNotInProgress
     case mobileAuthNotInProgress
     case authenticationNotInProgress
+    case pinCreationNotInProgress
+    case actionNotAllowed(description: String)
     
-    var description: String {
+    var errorDescription: String? {
```
Comment (kfa-onewelcome) [inline ios/OneginiWrapperErrors.swift:13]: why this is an optional?

File: ios/Handlers/RegistrationHandler.swift (hunk 1 above)
Comment (MichielVrins) [reply ios/Handlers/RegistrationHandler.swift:86]: They're slightly different, but I can move it to a seperate file where we store strings

Comment (kfa-onewelcome) [reply ios/Handlers/RegistrationHandler.swift:86]: oh I'm blind (didn't I mention it ;-)

File: ios/StringConstants.swift [hunk 3]
```diff
@@ -0,0 +1,2 @@
+let CancelCustomRegisrationNotAllowed = "Canceling the custom registration right now is not allowed. Registration is not in progress or pin creation has already started."
```
Comment (kfa-onewelcome) [inline ios/StringConstants.swift:1]: should be a static let, and starts from a lowercase

File: ios/StringConstants.swift [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class StringConstants {
```
Comment (kfa-onewelcome) [inline ios/StringConstants.swift:1]: the file is no longer needed :)

File: ios/Handlers/RegistrationHandler.swift [hunk 5]
```diff
@@ -76,14 +77,19 @@ extension RegistrationHandler : RegistrationConnectorToHandlerProtocol {
...
+            throw WrapperError.actionNotAllowed(description: RegistrationHandler.cancelBrowserRegisrationNotAllowed)
+        }
+        customRegistrationChallenge.sender.cancel(customRegistrationChallenge)
+        handleDidFailToRegister()
     }
 
     func cancelBrowserRegistration() throws {
         guard let browserRegistrationChallenge = self.browserRegistrationChallenge else {
-            throw WrapperError.registrationNotInProgress
+            throw WrapperError.actionNotAllowed(description: RegistrationHandler.cancelCustomRegisrationNotAllowed)
```
Comment (kfa-onewelcome) [inline ios/Handlers/RegistrationHandler.swift:89]: typo:
`cancelCustomRegisrationNotAllowed`
should be:
`cancelCustomRegistrationNotAllowed`

File: ios/Handlers/RegistrationHandler.swift [hunk 6]
```diff
@@ -76,14 +77,19 @@ extension RegistrationHandler : RegistrationConnectorToHandlerProtocol {
     }
 
     func cancelCustomRegistration() throws {
-        try handleOTPCode(nil, true)
+        guard let customRegistrationChallenge = self.customRegistrationChallenge else {
+            throw WrapperError.actionNotAllowed(description: RegistrationHandler.cancelBrowserRegisrationNotAllowed)
```
Comment (kfa-onewelcome) [inline ios/Handlers/RegistrationHandler.swift:81]: typo here as well

**Review by kfa-onewelcome — DISMISSED** [review]

**Review by aleknow — APPROVED** [review]

**Pull request merged**